src/pbt/prophecy_build_tool.py

```diff
@@ -0,0 +1,428 @@
+import collections
+from glob import glob
+import json
+import os
+import re
+import sys
+from typing import Dict
+
+import yaml
+from .process import Process
+from rich import print
+from databricks_cli.configure.config import _get_api_client
+from databricks_cli.configure.provider import EnvironmentVariableConfigProvider
+from databricks_cli.sdk import DbfsService, JobsService
+
+
+class ProphecyBuildTool:
+    def __init__(self, path_root: str):
+        print(
+            "[bold purple]Prophecy-build-tool[/bold purple] [bold black]v1.0.1[/bold black]\n"
+        )
+
+        self.path_root = path_root
+        self.path_project = os.path.join(self.path_root, "pbt_project.yml")
+
+        self._verify_project()
+        self._verify_databricks_configs()
+        self._parse_project()
+
+        config = EnvironmentVariableConfigProvider().get_config()
+
+        self.api_client = _get_api_client(config)
+
+        self.dbfs_service = DbfsService(self.api_client)
+        self.jobs_service = JobsService(self.api_client)
+
+        self.pipelines_build_path = {}
+
+    def build(self):
+        print("\n[bold blue]Building %s pipelines 🚰[/bold blue]" % self.pipelines_count)
+        overall_build_status = True
+        for pipeline_i, (path_pipeline, pipeline) in enumerate(self.pipelines.items()):
+            print(
+                "\n  Building pipeline %s [%s/%s]"
+                % (path_pipeline, pipeline_i + 1, self.pipelines_count)
+            )
+
+            path_pipeline_absolute = os.path.join(
+                os.path.join(self.path_root, path_pipeline), "code"
+            )
+            rc = (
+                self.build_python(path_pipeline_absolute)
+                if self.project_language == "python"
+                else self.build_scala(path_pipeline_absolute)
+            )
+
+            self.pipelines_build_path[path_pipeline] = None
+            build_file_found = False
+            build_file_paths = []
+
+            if self.project_language == "python":
+                path_pipeline_dist = path_pipeline_absolute + "/dist"
+                build_file_paths = list(
+                    filter(
+                        lambda x: (x.endswith("py3-none-any.whl")),
+                        glob(f"{path_pipeline_dist}/*.whl"),
+                    )
+                )
+            elif self.project_language == "scala":
+                path_pipeline_dist = path_pipeline_absolute + "/target"
+                build_file_paths = list(
+                    filter(
+                        lambda x: ("jar-with-dependencies" not in x),
+                        glob(f"{path_pipeline_dist}/*.jar"),
+                    )
+                )
```

Comment:
nit: maybe a else block which throws error/warning that language is undef. ?

Reply:
Already throws error if language is undefined during project parse itself